frontend/packages/kubevirt-plugin/integration-tests-cypress/view/wizard.ts

```diff
@@ -1,14 +1,70 @@
 import { ProvisionSource } from '../enums/provisionSource';
+import { VirtualMachineData } from '../types/vm';
+import { addDisk, addNIC } from './dialog';
+
+const fillBootSource = (
+  provisionSource: ProvisionSource,
+  pvcName: string,
+  pvcNS: string,
+  pvcSize?: string,
+) => {
+  cy.get('#image-source-type-dropdown').click();
+  cy.get('.pf-c-select__menu')
+    .contains(provisionSource.getDescription())
+    .click({ force: true });
+  switch (provisionSource) {
+    case ProvisionSource.URL: {
+      cy.get('input[id="provision-source-url"]').type(provisionSource.getSource());
+      if (pvcSize) {
+        cy.get('#request-size-input')
+          .clear()
+          .type(pvcSize);
+      }
+      break;
+    }
+    case ProvisionSource.CLONE_PVC: {
+      cy.get('body').then(($body) => {
+        if ($body.find('.pf-c-wizard__nav-list').length) {
+          cy.get('button[id="clone-pvc-ns"]').click();
+          cy.get(`a[id="${pvcNS}-Project-link"]`).click();
+          cy.contains('.pf-c-dropdown__toggle-text', 'Select Persistent Volume Claim').click({
+            force: true,
+          });
+          cy.get(`a[id="${pvcName}-PersistentVolumeClaim-link"]`).click();
+        } else {
+          cy.get('button[id="pvc-ns-dropdown"]').click();
+          cy.get(`a[id="${pvcNS}-Project-link"]`).click();
+          cy.get('button[id="pvc-name-dropdown"]').click();
+          cy.get(`a[id="${pvcName}-PersistentVolumeClaim-link"]`).click();
+        }
+      });
+      break;
+    }
+    case ProvisionSource.REGISTRY: {
+      cy.get('input[id="provision-source-container"]').type(provisionSource.getSource());
+      if (pvcSize) {
+        cy.get('#request-size-input')
+          .clear()
+          .type(pvcSize);
+      }
+      break;
+    }
+    default: {
+      break;
+    }
+  }
+};
 
 export const wizard = {
```

Comment:
I think this would be a good opportunity to take all of this out of the object and separate these into separate functions like we were talking about last week. Thoughts?

Reply:
There are lots of codes like this, let's do an overall review about it in a separate PR, wdyt?

Reply:
That works for me.